experimental/GITFans/GITFans.jl

```diff
@@ -0,0 +1,600 @@
+        if a[i].bits < b[i].bits
+            return true
+        elseif a[i].bits > b[i].bits
+            return false
+        end
+    end
+    return true
+end;
+
+##  Return a new bitset, consisting of the images of `bitset` under the
+##  permutation `perm`.
+function bitlist_oper(bitset::BitSet, perm::Vector{Int})
+    return_list = Int64[]
+    for i in bitset
+        push!(return_list, perm[i])
+    end
+    return BitSet(return_list)
+end;
+
+function bitlist_oper_tuple(bitset_tuple, perm_tuple)
+    return map(i -> bitlist_oper(bitset_tuple[i], perm_tuple[i]),
+               1:length(bitset_tuple))
+end;
+
+function find_smallest_orbit_element(elem, ggens, action, comparator, leq)
+    current_orbit = orbit(elem, ggens, action, comparator)
+    sorted_orbit = sort(current_orbit; lt=leq)
+    return sorted_orbit[1]
+end;
+
+function rewrite_action_to_orbits(homs)
+    G = domain(homs[1])
+    Ggens = gens(G)
+    generators_new_perm = [Vector{Int}[] for x in Ggens]
+
+    for hom in homs
+      for j in 1:length(generators_new_perm)
+        img = listperm(image(hom, Ggens[j]))
+        if length(img) == 0
+          img = Int[1]
+        end
+        push!(generators_new_perm[j], img)
+      end
+    end
+
+    return generators_new_perm
+end;
+
+
+#############################################################################
+
+# user functions
+
+"""
+    is_monomial_free(I::Oscar.MPolyIdeal, vars_to_zero = [])
+
+See Prop. 3.1 in [BKR](https://arxiv.org/abs/1603.09241).
+"""
+function is_monomial_free(I::Oscar.MPolyIdeal, vars_to_zero::Vector{Int} = Int[])
+    Oscar.singular_assure(I)
+    SingI = I.gens.S
+    R = Oscar.base_ring(SingI)
+    Rgens = Oscar.gens(R)
+    nr_variables = length(Rgens)
+    poly_list = map(f -> evaluate(f, vars_to_zero, fill(R(0), length(vars_to_zero))), gens(SingI))
+
+    # perm is an n-cycle
+    perm = collect(2:nr_variables)
+    push!(perm, 1)
+
+    Rgens_permuted = Rgens[perm]
+    si = Singular.Ideal(R, poly_list)
+
+    for i in 1:nr_variables
+        if !(nr_variables in vars_to_zero)
+            si = Singular.satstd( si, Singular.MaximalIdeal(R, 1))
+            if Singular.ngens(si) == 1 && si[1] == R(1)
+                return false
+            elseif reduce(R(1), si) == 0
+                return false
+            end
+        end
+        vars_to_zero = perm[vars_to_zero]
+        permuted_gens = [evaluate(j, Rgens_permuted) for j in gens(si)]
+        si = Singular.Ideal(R, permuted_gens)
+    end
+
+    return true
+end
+
+"""
+    orbit_cones(I::Oscar.MPolyIdeal, Q::Matrix{Int}, G::Oscar.GAPGroup = symmetric_group(1))
+
+> Return orbit representatives of the group `G` on the set of those cones
+> whose defining rays are given by subsets `S` of the rows of the matrix `Q`,
+> such that the matrix `S` has full rank and such that the ideal `I` is
+> monomial-free (see [`is_monomial_free`](@ref)) w.r.t. the variables `x_i`
+> for which the `i`-th row of `Q` is not contained in `S`.
```

Comment:
```suggestion
Return orbit representatives of the group `G` on the set of those cones
whose defining rays are given by subsets `S` of the rows of the matrix `Q`,
such that the matrix `S` has full rank and such that the ideal `I` is
monomial-free (see [`is_monomial_free`](@ref)) w.r.t. the variables `x_i`
for which the `i`-th row of `Q` is not contained in `S`.
```
and similarly elsewhere